src/JustSaying/Fluent/PublishConfig/DynamicAddressMessagePublisher.cs

```diff
@@ -102,6 +102,7 @@ public async Task PublishAsync(IEnumerable<Message> messages, PublishBatchMetada
                 var config = _staticConfigBuilder(topicArn);
 
                 var cachedPublisher = _batchPublisherCache.GetOrAdd(topicArn, config.BatchPublisher);
+                lockObj.Release(1);
```

Comment:
Should this be done in a finally block, I worry we could lose count and result in a hanging app

Reply:
Yeah, this makes sense. I've added a little extension so we can do this IDisposable-y